src/components/MultiSelect/MultiSelectListbox.tsx

```diff
@@ -0,0 +1,135 @@
+  onChange: (selectedItem: MultiSelectItem, id: string) => void;
+};
+
+function MultiSelectListbox({
+  id,
+  label,
+  defaultIsOpen,
+  items,
+  onChange,
+  selectedItems,
+  onClear,
+  width = "default",
+  isBlockElement = false,
+}: MultiSelectListboxProps) {
+  const styles = useMultiStyleConfig("MultiSelect", {
+    width,
+    isBlockElement,
+  });
+  // Downshift.
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({
+    items,
+    // Downshift's internal state for handling keyboard and mouse events.
+    stateReducer: (state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useSelect.stateChangeTypes.MenuKeyDownEnter:
+        case useSelect.stateChangeTypes.MenuKeyDownSpaceButton:
+        case useSelect.stateChangeTypes.ItemClick:
+          return {
+            ...changes,
+            isOpen: true, // Keep menu open after selection.
+            highlightedIndex: state.highlightedIndex,
+          };
+        default:
+          return changes;
+      }
+    },
+    // @ts-ignore
+    selectedItem: selectedItems,
+    onSelectedItemChange: ({ selectedItem }) => {
+      onChange(selectedItem, id);
+    },
+    initialIsOpen: defaultIsOpen,
+  });
+
+  return (
+    <Box id={id} __css={styles}>
+      <MultiSelectMenuButton
+        multiSelectId={id}
+        label={label}
+        isOpen={isOpen}
+        selectedItems={selectedItems}
+        onClear={onClear}
+        {...getToggleButtonProps({
+          // @TODO add short prefix to avoid id starting with number
+          id: `${id}-toggle-button`,
+          "aria-labelledby": `${id}-label ${id}-toggle-button`,
+        })}
+      />
+      <Box __css={styles.menuContainer} {...(!isOpen && { display: "none" })}>
+        <UnorderedList
+          styleType="none"
+          marginInlineStart="0"
+          {...getMenuProps({
+            // @TODO add short prefix to avoid id starting with number
+            id: `${id}-menu`,
+            "aria-labelledby": `${id}-label`,
+          })}
+          // @FIX This prevents the menu from closing when checkbox or label is clicked.
+          onClick={(e) => e.preventDefault()}
+          __css={styles.menu}
+        >
+          {isOpen &&
+            items.map((item: MultiSelectItem, index: number) => (
+              <ListItem
+                py={1}
+                // @TODO fix this, we want to pass the key prop as part of ...getItemProps but get
+                // error  Missing "key" prop for element in iterator  react/jsx-key
+                key={item.id}
+                {...getItemProps({
+                  // @TODO add short prefix to avoid id starting with number
+                  id: `${id}-item-${index}`,
+                  key: item.id,
+                  item,
+                  index,
+                })}
+                style={
```

Comment:
Can this be changed to `backgroundColor={highlightedIndex === index ? "#bde4ff" : undefined}`?
Where is the `#bde4ff` value coming from? This should be an existing DS color.

Reply:
I picked a random color that got left in the code. Might have to have @bigfishdesign13 weigh in on what color this should be, the highlight on "Art History", which shows if you hover or arrow down to via keyboard:

![ms-listbox-highlight](https://user-images.githubusercontent.com/62392174/171738429-e82fd158-9870-4168-858e-fc2fa6ad5a73.png)
.

Reply:
Yes, that color should be swapped with something from the palette.  Also, I’d like to see the spacing changed, which will take some adjustments to how the padding is applied to the different elements.  Please refer to URL below.
https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=60798%3A47074

I also see that we need to add a BG color to the default state of the `Checkbox`.

Reply:
This has been changed for the variant listbox. It does not seem to work for dialog because the logic for it to work in listbox is provided by downshift.